docs/layout-of-source-files.rst

```diff
@@ -216,101 +216,562 @@ the code below creates new global symbols ``alias`` and ``symbol2`` which refere
+       import "contracts/lib/token.sol";    // Relative to base
+       import "./contracts/lib/token.sol";  // Relative to source
+
+   There is actually no distinction between absolute imports and imports relative to base at the
+   virtual filesystem level - in both cases the import path is translated into an import key using
+   the same rules - but they are handled differently by the default file loader.
+
+   Imports relative to source, on the other hand, need to be interpreted as paths and normalized to
+   be properly resolved into import keys.
+   The path in this case must conform to UNIX path conventions regardless of the underlying platform.
+
+#. **CLI**
+
+   To compile a file using the command-line interface of the compiler you specify one or more paths:
+
+   .. code-block:: bash
+
+       solc contract.sol /usr/local/dapp-bin/token.sol
+
+   These are interpreted as *filesystem paths* and the rules for translating them into import keys
+   are different than for import paths.
+   They follow the shell rules instead: they can only be relatve to
+   the current working directory and they are normalized and converted into canonical form according
+   to platform-specific rules to determine which physical file they refer to.
+   Only if the file actually exists under that path, it is added to the virtual filesystem and the
+   import key is based on the canonical path.
+
+#. **Standard JSON (as content)**
+
+   An alternative way to compile your project is to use the ``--standard-json`` option and provide
+   a JSON file containing all of your source code:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "contract.sol": {
+                   "content": "import \"./util.sol\";\ncontract C {}"
+               },
+               "util.sol": {
+                   "content": "library Util {}"
+               },
+               "/usr/local/dapp-bin/token.sol": {
+                   "content": "contract Token {}"
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The ``sources`` dictionary specifies the initial content of the virtual filesystem and you
+   can use import keys directly there.
+   They do not undergo any extra translation or normalization.
+
+   The path to the JSON file does not affect the path resolution in any way.
+   In fact, it is common to supply it on the standard input in which case it does not have a path at all.
+
+   .. note::
+
+       When using ``--standard-json`` you cannot provide additional source files as command-line
+       arguments but it does not mean that the compiler will not load any extra files from disk.
+       If a contract imports a file that is not present in `sources`, the compiler will use the file
+       loader as in any other situation, which may result in the source being read from disk.
+
```

Comment:
I would also prefer this, but I think it was a user request. It might be very useful because it means that the compiler performs import resolution.

Reply:
Oh. I guess it's this one: #4688?

Maybe we could introduce a command-line option or something in `settings` in the JSON file so that you have to explicitly enable this?